mp/src/game/client/clientmode_shared.cpp

```diff
@@ -247,6 +248,8 @@ ClientModeShared::ClientModeShared()
 	m_pChatElement = NULL;
 	m_pWeaponSelection = NULL;
 	m_nRootSize[ 0 ] = m_nRootSize[ 1 ] = -1;
+
+    m_pChatSafeguard = new MomRunSafeguard("open chat");
```

Comment:
I was mainly doing this to avoid having the import the file, as forward declaring this without using a pointer was throwing errors. When compiling sources from other files that use, say, `CMOMSaveLocSystem`, it says it uses an undefined class.

Reply:
You can shift actually defining the safeguards to a safeguard system that houses all the convars as well, and then anything that wants to check it can just do like `g_pSafeguards->IsSafeguarded(SafeguardType_t type)`

Reply:
Ooooo and here I'm assuming by type you mean chat/practice/restart/etc?

Reply:
Precisely.